pkg/issuer/acme/dns/util/dns.go

```diff
@@ -1,9 +1,20 @@
 package util
 
-import "fmt"
+import (
+	"fmt"
+
+	"github.com/miekg/dns"
+)
 
 // DNS01Record returns a DNS record which will fulfill the `dns-01` challenge
 // TODO: move this into a non-generic place by resolving import cycle in dns package
 func DNS01Record(domain, value string) (string, string, int) {
-	return fmt.Sprintf("_acme-challenge.%s.", domain), value, 60
+	fqdn := fmt.Sprintf("_acme-challenge.%s.", domain)
+
+	// Check if the domain has CNAME then return that
+	r, err := dnsQuery(fqdn, dns.TypeCNAME, RecursiveNameservers, true)
```

Comment:
RecursiveNameservers does not respect the recently added `--dns01-self-check-nameservers` flag. We should plumb this into this point too imo.